extension/src/Runner/testRunner.ts

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { ITestRunnerContext } from "./testRunnerContext";
+
+export interface ITestRunner {
+    run(context: ITestRunnerContext): undefined | Promise<undefined>;

```

Comment:
Returning `undefined` will cost the caller an `if` check to handle the return value differently. Why not just use Promise and reject when there's something wrong?

Reply:
The interface is just too generic. It is not clear enough to the caller about what to provide and what to expect. Please rework this. You could have stages like `beforeRun`, `run` and `postRun` and each have clear definitions of what to implement.

Reply:
make sense